Source/DafnyCore/AST/Modules/ModuleDefinition.cs

```diff
@@ -58,12 +57,8 @@ public class ModuleDefinition : RangeNode, IDeclarationOrUsage, IAttributeBearin
   public readonly List<TopLevelDecl> ResolvedPrefixNamedModules = new();
   [FilledInDuringResolution]
   public readonly List<PrefixNameModule> PrefixNamedModules = new();  // filled in by the parser; emptied by the resolver
-  public virtual IEnumerable<TopLevelDecl> TopLevelDecls =>
-    defaultClassFirst ? DefaultClasses.
+  public virtual IEnumerable<TopLevelDecl> TopLevelDecls => DefaultClasses.
```

Comment:
I like this this PR simplifies the code, but Translator shouldn't depend on the order of TopLevelDecls, should it?

Reply:
The order does not matter, except for one thing: The default declaration should be first, according to the current optimization of Boogie pruning.
Before the pruning of unused declarations in boogie, the order surely did not matter.

Reply:
How is the pruning affected by order? I don't think it should. Is that a bug?

Reply:
The pruning is affected by order because of this condition here:

https://github.com/dafny-lang/dafny/blob/2b26b9d1cbb228a5bc47df5b1e0acb64287ec8b3/Source/DafnyCore/Verifier/Translator.ClassMembers.cs#L165-L169

When the default class is first, it will populate the `referenceMembers` for every definition that is referenced, so even if the method is in another file (!InVerificationScope), that method will still be added.
When the default class is not first, the `referenceMembers` will be empty when dealing with the dependencies first, so the method is ignored.

I don't have any more bandwidth in the next weeks to take care of this issue. Would you know of a quicker fix?